pkg/generate/app/app.go

```diff
@@ -138,23 +140,26 @@ func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTrigge
 			},
 		},
 	}
-	var source *buildapi.BuildSource
-	switch {
-	case r.URL != nil:
-		source = &buildapi.BuildSource{
-			Type: buildapi.BuildSourceGit,
-			Git: &buildapi.GitBuildSource{
-				URI: urlWithoutRef(*r.URL),
-				Ref: r.Ref,
-			},
-			ContextDir: r.ContextDir,
-		}
-	case len(r.DockerfileContents) != 0:
+	source := &buildapi.BuildSource{}
+
+	if len(r.DockerfileContents) != 0 {
```

Comment:
Comments addressed, please rereview

On Tue, Oct 27, 2015 at 4:12 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> The types are not mutually exclusive.
>
> On Tue, Oct 27, 2015 at 2:02 PM, Cesar Wong <notifications@github.com>
> wrote:
>
>> In pkg/generate/app/app.go
>> <https://github.com/openshift/origin/pull/5411#discussion_r43161093>:
>>
>> > @@ -138,23 +140,26 @@ func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTrigge
>> >  			},
>> >  		},
>> >  	}
>> > -	var source *buildapi.BuildSource
>> > -	switch {
>> > -	case r.URL != nil:
>> > -		source = &buildapi.BuildSource{
>> > -			Type: buildapi.BuildSourceGit,
>> > -			Git: &buildapi.GitBuildSource{
>> > -				URI: urlWithoutRef(*r.URL),
>> > -				Ref: r.Ref,
>> > -			},
>> > -			ContextDir: r.ContextDir,
>> > -		}
>> > -	case len(r.DockerfileContents) != 0:
>> > +	source := &buildapi.BuildSource{}
>> > +
>> > +	if len(r.DockerfileContents) != 0 {
>>
>> shouldn't this still be a switch since the different types are mutually
>> exclusive?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/5411/files#r43161093>.
>>
>
>